### PR TITLE
[WIP]RHDEVDOCS-2738 Document builds: Record Last Triggered Image ID in status

### PR DESCRIPTION
**Pull request by rolfedh**
- Aligned team: Dev Tools
- For branches: master, 4.8
- Jira: https://issues.redhat.com/browse/RHDEVDOCS-2738
- Direct link to doc preview: 
- SME review: @gabemontero 
- QE review: @xiaoping378 
- Peer review: @neal-timpe 
- <All reviews complete. Please merge now>

### CODE REVIEW

File: rest_api/workloads_apis/buildconfig-build-openshift-io-v1.adoc [hunk 1]
```diff
@@ -1200,7 +1200,9 @@ Type::
 
 | `lastTriggeredImageID`
 | `string`
-| lastTriggeredImageID is used internally by the ImageChangeController to save last used image ID for build
+| lastTriggeredImageID is used internally by the ImageChangeController to save last used image ID for build. Deprecated. {product-title} version 4.9 will remove support for `lastTriggeredImageID` and ignore it. Then, image change triggers will not start a build based on changes to the `lastTriggeredImageID` field in the `BuildConfig` spec. Instead, the image IDs that trigger a build will be recorded in the status of the `BuildConfig` object, which most users cannot change.
+
+Therefore, update scripts and jobs that inspect `checkingbuildConfig.spec.triggers[i].imageChange.lastTriggeredImageID` accordingly. (link:https://issues.redhat.com/browse/BUILD-213[*BUILD-213*])
```
Comment (rolfedh) [inline rest_api/workloads_apis/buildconfig-build-openshift-io-v1.adoc:1205]: @jboxman I'm noticing some `..` formatting throughout this file, like you see on line 1210 below, and wondering if these should be H2 or H<n> titles instead. WDYT?

Comment (gabemontero) [reply rest_api/workloads_apis/buildconfig-build-openshift-io-v1.adoc:1205]: formatting aside @rolfedh the text for the spec field looks good here.

**Comment by gabemontero** [comment]
OK @rolfedh when I bring up 

I only see this for the status:

```
<div>
<div>.status</div>
<dl>
<dt>Description</dt>
<dd>
<p>BuildConfigStatus contains current state of the build config object.</p>
</dd>
<dt>Type</dt>
<dd>
<p><code>object</code></p>
</dd>
<dt>Required</dt>
<dd>
<div>
<ul>
<li>
<p><code>lastVersion</code></p>
</li>
</ul>
</div>
</dd>
</dl>
<div>
<div>.status</div>
<dl>
<dt>Description</dt>
<dd>
<p>BuildConfigStatus contains current state of the build config object.</p>
</dd>
<dt>Type</dt>
<dd>
<p><code>object</code></p>
</dd>
<dt>Required</dt>
<dd>
<div>
<ul>
<li>
<p><code>lastVersion</code></p>
</li>
</ul>
</div>
</dd>
</dl>
</div>


Property | Type | Description
-- | -- | --
lastVersion | integer | lastVersion is used to inform about number of last triggered build.


</div>


Property | Type | Description
-- | -- | --
lastVersion | integer | lastVersion is used to inform about number of last triggered build.


```

Sorry, github does not allow me to point to a line in that adoc file.  If I select edit in that file, it seems the status section starts at line 1209 with what is currently up in the master branch.

But I was expecting our new status fields to follow what you all have there for lastVersion.  I don't see that in this PR.

By comparison, look at the actual go code:  https://github.com/openshift/api/blob/master/build/v1/types.go#L994-L1002

**Comment by jboxman** [comment]
cc @ctauchen, looks like people are noticing this isn't regenerated for 4.8 yet; that's my fault.

**Comment by rolfedh** [comment]
> I only see this for the status:

@gabemontero. @jboxman is going to regenerate the API docs topics for builds from the 4.8 branch this week. The current godoc comments will appear there.

**Pull request closed without merging**